### ISSUE TITLE
Can't build armv6l-unknown-linux-gnueabihf-ghc (or -perl) on macOS with release-19.03

### ISSUE THREAD

**Issue by angerman** [NixOS/nixpkgs#67433]
**Describe the bug**
Can't build arm cross compilers.

**To Reproduce**
`../nixpkgs` set to release-19.03

```
nix build '(with (import ../nixpkgs { crossSystem = { config = "armv6l-unknown-linux-gnueabihf"; }; }); haskell.compiler.ghc865)' --show-trace  ~/Projects/iohk/haskell.nix
warning: Nix search path entry '/nix/var/nix/profiles/per-user/root/channels' does not exist, ignoring
error: while evaluating the attribute 'buildInputs' of the derivation 'ghc-8.6.5-armv6l-unknown-linux-gnueabihf' at /Users/angerman/Projects/iohk/nixpkgs/pkgs/stdenv/generic/make-derivation.nix:196:11:
while evaluating the attribute 'configureFlags' of the derivation 'perl-5.28.1-armv6l-unknown-linux-gnueabihf' at /Users/angerman/Projects/iohk/nixpkgs/pkgs/stdenv/generic/make-derivation.nix:196:11:
while evaluating 'getOutput' at /Users/angerman/Projects/iohk/nixpkgs/lib/attrsets.nix:464:23, called from /Users/angerman/Projects/iohk/nixpkgs/pkgs/development/interpreters/perl/default.nix:22:13:
while evaluating the attribute 'cc.libc' at /Users/angerman/Projects/iohk/nixpkgs/pkgs/stdenv/generic/default.nix:148:14:
while evaluating the attribute 'gcc' at /Users/angerman/Projects/iohk/nixpkgs/pkgs/top-level/all-packages.nix:6857:3:
while evaluating 'addMetaAttrs' at /Users/angerman/Projects/iohk/nixpkgs/lib/meta.nix:15:28, called from /Users/angerman/Projects/iohk/nixpkgs/pkgs/top-level/all-packages.nix:6999:10:
while evaluating 'wrapCC' at /Users/angerman/Projects/iohk/nixpkgs/pkgs/top-level/all-packages.nix:7776:12, called from /Users/angerman/Projects/iohk/nixpkgs/pkgs/top-level/all-packages.nix:6999:19:
while evaluating 'wrapCCWith' at /Users/angerman/Projects/iohk/nixpkgs/pkgs/top-level/all-packages.nix:7755:5, called from /Users/angerman/Projects/iohk/nixpkgs/pkgs/top-level/all-packages.nix:7776:16:
while evaluating 'callPackageWith' at /Users/angerman/Projects/iohk/nixpkgs/lib/customisation.nix:108:35, called from /Users/angerman/Projects/iohk/nixpkgs/pkgs/top-level/all-packages.nix:7764:7:
while evaluating 'makeOverridable' at /Users/angerman/Projects/iohk/nixpkgs/lib/customisation.nix:67:24, called from /Users/angerman/Projects/iohk/nixpkgs/lib/customisation.nix:112:8:
while evaluating anonymous function at /Users/angerman/Projects/iohk/nixpkgs/pkgs/build-support/cc-wrapper/default.nix:8:1, called from /Users/angerman/Projects/iohk/nixpkgs/lib/customisation.nix:69:12:
while evaluating the attribute 'postInstall' of the derivation 'glibc-2.27-armv6l-unknown-linux-gnueabihf' at /Users/angerman/Projects/iohk/nixpkgs/pkgs/stdenv/generic/make-derivation.nix:196:11:
while evaluating the attribute 'buildInputs' of the derivation 'glibc-2.27' at /Users/angerman/Projects/iohk/nixpkgs/pkgs/development/libraries/glibc/common.nix:154:3:
while evaluating the attribute 'ARCH' of the derivation 'linux-headers-4.19.16' at /Users/angerman/Projects/iohk/nixpkgs/pkgs/os-specific/linux/kernel-headers/default.nix:8:5:
while evaluating the attribute 'hostPlatform.platform.kernelArch' at /Users/angerman/Projects/iohk/nixpkgs/lib/systems/default.nix:26:7:
missing kernelArch
```
or
```
nix build '(with (import ../nixpkgs { crossSystem = { config = "armv6l-unknown-linux-gnueabihf"; }; }); perl)' --show-trace                     ~/Projects/iohk/haskell.nix
warning: Nix search path entry '/nix/var/nix/profiles/per-user/root/channels' does not exist, ignoring
error: while evaluating the attribute 'configureFlags' of the derivation 'perl-5.28.1-armv6l-unknown-linux-gnueabihf' at /Users/angerman/Projects/iohk/nixpkgs/pkgs/stdenv/generic/make-derivation.nix:196:11:
while evaluating 'getOutput' at /Users/angerman/Projects/iohk/nixpkgs/lib/attrsets.nix:464:23, called from /Users/angerman/Projects/iohk/nixpkgs/pkgs/development/interpreters/perl/default.nix:22:13:
while evaluating the attribute 'cc.libc' at /Users/angerman/Projects/iohk/nixpkgs/pkgs/stdenv/generic/default.nix:148:14:
while evaluating the attribute 'gcc' at /Users/angerman/Projects/iohk/nixpkgs/pkgs/top-level/all-packages.nix:6857:3:
while evaluating 'addMetaAttrs' at /Users/angerman/Projects/iohk/nixpkgs/lib/meta.nix:15:28, called from /Users/angerman/Projects/iohk/nixpkgs/pkgs/top-level/all-packages.nix:6999:10:
while evaluating 'wrapCC' at /Users/angerman/Projects/iohk/nixpkgs/pkgs/top-level/all-packages.nix:7776:12, called from /Users/angerman/Projects/iohk/nixpkgs/pkgs/top-level/all-packages.nix:6999:19:
while evaluating 'wrapCCWith' at /Users/angerman/Projects/iohk/nixpkgs/pkgs/top-level/all-packages.nix:7755:5, called from /Users/angerman/Projects/iohk/nixpkgs/pkgs/top-level/all-packages.nix:7776:16:
while evaluating 'callPackageWith' at /Users/angerman/Projects/iohk/nixpkgs/lib/customisation.nix:108:35, called from /Users/angerman/Projects/iohk/nixpkgs/pkgs/top-level/all-packages.nix:7764:7:
while evaluating 'makeOverridable' at /Users/angerman/Projects/iohk/nixpkgs/lib/customisation.nix:67:24, called from /Users/angerman/Projects/iohk/nixpkgs/lib/customisation.nix:112:8:
while evaluating anonymous function at /Users/angerman/Projects/iohk/nixpkgs/pkgs/build-support/cc-wrapper/default.nix:8:1, called from /Users/angerman/Projects/iohk/nixpkgs/lib/customisation.nix:69:12:
while evaluating the attribute 'postInstall' of the derivation 'glibc-2.27-armv6l-unknown-linux-gnueabihf' at /Users/angerman/Projects/iohk/nixpkgs/pkgs/stdenv/generic/make-derivation.nix:196:11:
while evaluating the attribute 'buildInputs' of the derivation 'glibc-2.27' at /Users/angerman/Projects/iohk/nixpkgs/pkgs/development/libraries/glibc/common.nix:154:3:
while evaluating the attribute 'ARCH' of the derivation 'linux-headers-4.19.16' at /Users/angerman/Projects/iohk/nixpkgs/pkgs/os-specific/linux/kernel-headers/default.nix:8:5:
while evaluating the attribute 'hostPlatform.platform.kernelArch' at /Users/angerman/Projects/iohk/nixpkgs/lib/systems/default.nix:26:7:
missing kernelArch
```
**Expected behavior**
build perl or ghc as crosscompilers

**Metadata**
Please run `nix run nixpkgs.nix-info -c nix-info -m` and paste the result.
```
 - system: `"x86_64-darwin"`
 - host os: `Darwin 18.7.0, macOS 10.14.6`
 - multi-user?: `no`
 - sandbox: `no`
 - version: `nix-env (Nix) 2.2.2`
 - channels(angerman): `"darwin, nixpkgs-19.09pre186574.88d9f776091"`
warning: Nix search path entry '/nix/var/nix/profiles/per-user/root/channels' does not exist, ignoring
 - nixpkgs: `/Users/angerman/.nix-defexpr/channels/nixpkgs`
```

Maintainer information:
```yaml
# a list of nixpkgs attributes affected by the problem
attribute:
# a list of nixos modules affected by the problem
module:
```

**Comment by angerman** [comment]
Git bisect yields: ff1b945851f778cb0c8f42faf5d5f4d931912e13

**Comment by angerman** [comment]
master is fixed because it contains 00a41ebbc15b2345819612b8f917e1d69d853f5a

I hereby suggest we merge 00a41ebbc15b2345819612b8f917e1d69d853f5a into release-19.03 as well!

**Comment by matthewbauer** [comment]
Cherry picked in 3924b344254350e4bee875a3353c1093e325743f
